### PR TITLE
support autoxgbclassifier and add more features in raytune search engine

### PR DESCRIPTION
**Pull request by dding3**
1. support autoxgbclassifier
2. support SkOpt search
3. support AsyncHyperBand scheduler
4. add accuracy metric
5. add autoxgbclassifier example
6. bug fix for get_best_config()

### CODE REVIEW

**Comment by dding3** [comment]
jekins pass. http://10.239.47.210:18888/job/ZOO-PR-Validation/4276/

**Comment by dding3** [comment]
Merge this pr first, will create a new pr to fix if there is comments about this pr.

File: pyzoo/zoo/automl/model/__init__.py [hunk 1]
```diff
@@ -13,3 +13,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from .abstract import *
+from .VanillaLSTM import *
+from .Seq2Seq import *
+from .time_sequence import *
+from .tcmf_model import *
+from .XGBoost import *
```
Comment (yushan111) [inline pyzoo/zoo/automl/model/__init__.py:21]: I have removed those imports to avoid unnecessary installation of other models' dependencies.

Comment (yushan111) [reply pyzoo/zoo/automl/model/__init__.py:21]: jenkins fail due to these changes. I will open another PR to fix.

**Pull request merged**